packages/wrangler/src/config/validation.ts

```diff
@@ -1299,12 +1299,20 @@ function normalizeAndValidateEnvironment(
 			topLevelEnv,
 			rawEnv,
 			"logfwdr",
-			validateBindingsProperty(envName, validateCflogfwdrBinding),
+			validateCflogfwdrObject(envName),
 			{
-				schema: undefined,
+				capnp_schema: undefined,
 				bindings: [],
 			}
 		),
+		capnp_src_prefix: inheritable(
+			diagnostics,
+			topLevelEnv,
+			rawEnv,
+			"capnp_src_prefix",
+			isString,
+			undefined
+		),
```

Comment:
I am not very keen on this being a top level public setting in wrangler.toml.
Apart from anything, is it possible that unsafe capnp bindings might want a different prefix to the logfwdr ones?

Can we move it into the places where it is needed. I.e. inside the logfwdr setting and inside the unsafe capnp binding?
I appreciate that this could make things more verbose for those using capnp unsafe bindings but I feel it is cleaner and safer.

Reply:
The issue is that we can only specify this once for all capnp schemas we're compiling - if we move it into another place, we either:
- have to mandate it be in the logfwdr section, which doesn't really make sense if you're only using an unsafe binding (or vice versa)
- have to require it in every place, which also doesn't really make sense

I didn't really like putting it in the top level env either, but it's the only place that really made sense.

I suppose one other option would be refactoring all of this to add one option somewhere, and not putting it within `unsafe.bindings` *or* in `logfwdr`, something like this
```toml
[capnp]
src_prefix = "./foo"
schemas = [ "./my-schema", "./my-other-schema" ]
```

Reply:
> have to require it in every place, which also doesn't really make sense

This is my recommendation.

Reply:
We'd also have to check they're all resolving to the same place then - which is possible, just feels a little odd 😅

Reply:
But what if someone wants to remove a different prefix for each unsafe capnp binding? I don't think that we should assume they would always be the same.

Reply:
And why would they have to resolve to the same place in any case?

Reply:
That would require changes to capnp itself - we can only pass through one source prefix as far as I can tell

Reply:
Hmm... interesting.
I forgot that you are only running the capnp tool once but concatenating all the capnp source files into a single file.
This actually feels rather brittle to me.
Instead I would recommend that you create a new "entry-point" source file that "imports" all the files that are included in the config. Is that feasible?

Reply:
Hmm, do you have an example of that? Just to make sure I understand correctly

Reply:
Discussed offline - we are going to go with a new `[unsafe_capnp]` section in wrangler.toml as described above.

Reply:
This should be resolved now!